src/graph/planner/plan/Query.h

```diff
@@ -801,10 +799,9 @@ class Sample final : public SingleInputNode {
   }
 
   // Get constant count
-  int64_t count() const {
-    DCHECK(ExpressionUtils::isEvaluableExpr(count_));
-    QueryExpressionContext qec;
-    auto count = count_->eval(qec).getInt();
+  int64_t count(const QueryContext* qctx) const {
+    DCHECK(ExpressionUtils::isEvaluableExpr(count_, qctx));
+    auto count = count_->eval(QueryExpressionContext(qctx_->ectx())()).getInt();
```

Comment:
Why change this?

Reply:
You are right, there is no need to support stateful checking on `Sample` plan node, I will change it back.